### DISCUSSION TITLE
How do I display the filename instead of the id with SpatieMediaLibraryFileUpload?

### DISCUSSION THREAD

**Discussion by l-yk-l** [filamentphp/filament#3896] [Help]
I use SpatieMediaLibraryFileUpload and I need to display the filename instead of id of my media in form. How can I do that?
![image](https://user-images.githubusercontent.com/33335112/188846041-965adb51-31f6-427c-a3b6-83004aee414a.png)

Thank you in advance

**Comment by danharrin** [comment]
Thats very weird... any console errors?

**Reply by l-yk-l** [reply]
Only one

![image](https://user-images.githubusercontent.com/33335112/188866381-5454b1ec-cce1-452a-a17f-930cf5ee006e.png)

```php
SpatieMediaLibraryFileUpload::make('filenames')
                    ->visibility('private')
                    ->collection('lesson')
                    ->acceptedFileTypes(['application/pdf'])
                    ->multiple()
                    ->maxSize(1024*3)
                    ->disablePreview()
                    ->enableDownload()
                    ->preserveFilenames(),
```
This is how I use this field

**Reply by danharrin** [reply]
Please run `filament:upgrade` and remove any files from /resources/views/vendor, that could fix the console error. Not sure why the file name is wrong on the upload though

**Reply by l-yk-l** [reply]
It didn't help :(

**Reply by l-yk-l** [reply]
The file name is displayed when creating my model, but not when viewing and editing

**Reply by danharrin** [reply]
just checking, there is no `filenames` column in your database, right?

**Reply by l-yk-l** [reply]
right

**Reply by danharrin** [reply]
What does your media table contents look like

**Reply by l-yk-l** [reply]
I don't know how to show it to you
That maybe?
```php
[
	"id" => 1,
	"model_type" => "App\Domains\Teaching\Models\Lesson",
	"model_id" => 4,
	"uuid" => "38440319-4e04-40e7-859a-6f765zxc24dc",
	"collection_name" => "lesson",
	"name" => "629-1",
	"file_name" => "629-1.pdf",
	"mime_type" => "application/pdf",
	"disk" => "public",
	"conversions_disk" => "public",
	"size" => "343510",
	"manipulations" => "[]",
	"custom_properties" => "[]",
	"generated_conversions" => "[]",
	"responsive_images" => "[]",
	"order_column" => 1,
	"created_at" => "2022-08-18 02:54:32",
	"updated_at" => "2022-08-18 02:54:32",
]
```

**Reply by danharrin** [reply]
where is 7 from?

**Reply by l-yk-l** [reply]
it is id of 7th media record

**Reply by l-yk-l** [reply]
I send first record as example

**Reply by danharrin** [reply]
I don't know then, something is wrong in your project. I don't know how to help you unless you send a reproduction repo or something, with a full database seeder and everything

**Reply by l-yk-l** [reply]
Thank you for your time, it's very valuable to me
I'll try to start over :(

**Reply by l-yk-l** [reply]
problem was with my custom urlGenerator

This returned me a link with the model ID, and then this ID was shown as the file name.
